### PR TITLE
add BootProgress property to redfish_info

### PR DESCRIPTION
**Pull request by danxg87**
##### SUMMARY

Adding the BootProgress field to redfish_info systems properties. This should expose things like LastState to determine if the host reported booting into OS OR if there were boot failures/issues. This will be available on hosts running Redfish 1.13+ based on the DMTF documentation and should prove useful for checking the boot state/status of hosts post-reboot/reset.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

redfish_info
##### ADDITIONAL INFORMATION

```paste below
"BootProgress": {
                "LastState": "OSRunning"
            },
```

### CODE REVIEW

File: changelogs/fragments/7626-redfish-info-add-boot-progress-property.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - redfish_utils.py - adding BootProgress field to Systems inventory properties. This will add this data for hosts running Redfish 1.13+ per DMTF docs.
```
Comment (felixfontein) [inline changelogs/fragments/7626-redfish-info-add-boot-progress-property.yml:2]: Could you please adjust this so it tells a user what actually changes? The readers of this changelog are not collection developers, but end-users who use modules/plugins from the collection. So something like:
```
minor_changes:
  - redfish_info - ... (https://github.com/ansible-collections/community.general/pull/7626).
```
Thanks!

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution!

File: changelogs/fragments/7626-redfish-info-add-boot-progress-property.yml (hunk 1 above)
Comment (danxg87) [reply changelogs/fragments/7626-redfish-info-add-boot-progress-property.yml]: Thanks for that! I've tweaked the description to highlight the module and where this change happens.

**Comment by mraineri** [comment]
Looks good to me; thanks!

File: changelogs/fragments/7626-redfish-info-add-boot-progress-property.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - redfish_info - adding the BootProgress property when getting Systems info (https://github.com/ansible-collections/community.general/pull/7626)
```
Comment (felixfontein) [inline changelogs/fragments/7626-redfish-info-add-boot-progress-property.yml:2]:
```suggestion
  - redfish_info - adding the ``BootProgress`` property when getting ``Systems`` info (https://github.com/ansible-collections/community.general/pull/7626).
```

**Comment by felixfontein** [comment]
@danxg87 thanks for your contribution!
@mraineri thanks for reviewing!

**Pull request merged**